gci/models.py

```diff
@@ -1 +1,107 @@
-# Create your models here.
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+from django.db import models
+
+
+class Student(models.Model):
+
+    identifier = models.IntegerField(default=0, primary_key=True)
+    name = models.CharField(max_length=100)
+    program_year = models.ForeignKey('ProgramYear', on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.name
+
+
+class Organization(models.Model):
+
+    identifier = models.IntegerField(default=0, primary_key=True)
+    name = models.CharField(max_length=100)
+    logo = models.ImageField(upload_to='', blank=True)
+    url = models.URLField(blank=True)
+    summary = models.TextField(max_length=300, default='')
+    students = models.ManyToManyField(Student)
+
+    def __str__(self):
+        return self.name
+
+
+class Task(models.Model):
+
+    task_status = (
+        (1, 'Draft'),
+        (2, 'Published'),
+    )
+
+    name = models.CharField(max_length=200)
+    description = models.TextField(max_length=500)
+    status = models.IntegerField(choices=task_status, default=1)
+    max_instances = models.IntegerField(default=1)
+    mentors = models.ManyToManyField('Mentor')
+    tags = models.ManyToManyField('Tag')
+    is_beginner = models.BooleanField(default=False)
+    categories = models.ManyToManyField('Category')
+    time_to_complete_in_days = models.IntegerField(default=3)
+    external_url = models.URLField()
+    metadata = models.TextField()
+    last_modified = models.DateField()
+    org = models.ForeignKey(Organization, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.name
+
+
+class TaskInstance(models.Model):
+
+    task_instance = (
+        (1, 'CLAIMED'),
+        (2, 'ABANDONED'),
+        (3, 'SUBMITTED'),
+        (4, 'NEEDS_WORK'),
+        (5, 'OUT_OF_TIME'),
+        (6, 'COMPLETED'),
+    )
+
+    task = models.ForeignKey(Task, on_delete=models.CASCADE)
+    student = models.ForeignKey(Student, on_delete=models.CASCADE)
+    organization = models.ForeignKey(Organization, on_delete=models.CASCADE)
+    status = models.IntegerField(choices=task_instance)
+    completion_date = models.DateField()
+    deadline = models.DateField()
+    modified = models.DateField()
+
+    def __str__(self):
+        return self.task
```

Comment:
IMO it'd good to have student name here too, might get confusing for tasks with multiple instances.

Reply:
Right, task + student